internal/backend/telemetry/telemetry.go

```diff
@@ -0,0 +1,128 @@
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/configset"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/metric"
+	noop "go.opentelemetry.io/otel/metric/noop"
+	sdk "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+	"go.uber.org/zap"
+)
+
+type Config struct {
+	Enabled     bool   `koanf:"enabled"`
+	ServiceName string `koanf:"service_name"`
+	Endpoint    string `koanf:"endpoint"`
+}
+
+var Configs = configset.Set[Config]{
+	Default: &Config{Enabled: true, ServiceName: "ak", Endpoint: "localhost:4318"},
+	Dev:     &Config{Enabled: false, ServiceName: "ak", Endpoint: "localhost:4318"},
+}
+
+func (cfg *Config) fixConfig() {
+	if cfg.ServiceName == "" {
+		cfg.ServiceName = Configs.Default.ServiceName
+	}
+	if cfg.Endpoint == "" {
+		cfg.Endpoint = Configs.Default.Endpoint
+	}
+}
+
+func WithLabels(args ...string) metric.MeasurementOption {
+	var attrs []attribute.KeyValue
+	if len(args)%2 != 0 {
+		args = args[:len(args)-1] // strip the last one. TODO: log?
+	}
+	for i := 0; i < len(args); i += 2 {
+		attrs = append(attrs, attribute.String(args[i], args[i+1]))
+	}
+	return metric.WithAttributes(attrs...)
+}
+
+type Telemetry struct {
+	l   *zap.Logger
+	cfg Config
+}
+
+func New(z *zap.Logger, cfg *Config) *Telemetry {
+	cfg.fixConfig() // just ensure that endpoint and service name are set
+
+	telemetry := &Telemetry{l: z, cfg: *cfg}
+
+	if !telemetry.cfg.Enabled {
+		z.Info("metrics are disabled")
+		return telemetry
+	}
+
+	// TODO: [ENG-1445] GRPC?
```

Comment:
I think the proper format is `// TODO(ENG-1445): gRPC?`.

Reply:
done